lib/stanford-mods/imprint.rb

```diff
@@ -37,8 +37,44 @@ def display_str
         imprint_statements.join('; ') if imprint_statements.present?
       end
 
+      def publication_date_for_slider
+        @originInfo_ng_nodeset.map do |origin_info_node|
+
+          date_elements = if origin_info_node.as_object.first.key_dates.any?
+                            origin_info_node.as_object.first.key_dates.map(&:as_object).map(&:first)
+                          else
+                            date_field_keys.map do |date_field|
+                              next unless origin_info_node.respond_to?(date_field)
+                              date_elements = origin_info_node.send(date_field)
+                              date_elements.map(&:as_object).map(&:first) if date_elements.any?
+                            end.first
+                          end
+
+          if date_elements.none?
+            []
+          elsif date_elements.find(&:start?) && date_elements.find(&:end?)
+            start_date = date_elements.find(&:start?)
+            end_date = date_elements.find(&:end?)
+
+            (start_date.as_range.min.year..end_date.as_range.max.year).to_a
+          elsif date_elements.find(&:start?)
+            start_date = date_elements.find(&:start?)
+
+            (start_date.as_range.min.year..Time.now.year).to_a
+          elsif date_elements.one?
+            date_elements.first.to_a.map(&:year)
+          else
+            date_elements.map { |v| v.to_a.map(&:year) }.flatten
```

Comment:
Can you add a comment here about what you are expecting `date_elements` to look like when we hit this branch?

Reply:
Literally anything else?

Reply:
Is that so?  It looks to me like you are expecting an enumerable that is populated with methods that respond to `#to_a`?

Reply:
That's true -- but this method is producing that for us.